UnityProject/Assets/Scripts/Clothing/Eyewear/NightVisionGoggles.cs

```diff
@@ -3,16 +3,36 @@
 using Player;
 using UI.Action;
 
-public class NightVisionGoggles : NetworkBehaviour, IItemInOutMovedPlayerButClientTracked,
+public class NightVisionGoggles : NetworkBehaviour, IItemInOutMovedPlayer,
 	ICheckedInteractable<HandActivate>
 {
+
+	private NightVisionData VisionData = new NightVisionData(b:true);
+
 	[SerializeField, Tooltip("How far the player will be able to see in the dark while he has the goggles on.")]
-	private Vector3 nightVisionVisibility;
+	private Vector3 nightVisionVisibility { get; set; }
+
+	[System.Serializable]
+	public struct NightVisionData
+	{
+		public bool isOn  { get; set; }
+
+
+		[SerializeField, Tooltip("How far the player will be able to see in the dark while he has the goggles on.")]
+		public Vector3 nightVisionVisibility { get; set; }
+
+		[SerializeField, Tooltip("How fast will the player gain visibility?")]
+		public float visibilityAnimationSpeed { get; set; }
+
+		public NightVisionData(bool b)
+		{
+			isOn = false;
+			nightVisionVisibility = new Vector3(10.5f, 10.5f, 21);
+			visibilityAnimationSpeed = 1.5f;
+		}
```

Comment:
b is not used?

Reply:
for some reason the default initialiser didn't work